files/default/pcluster_connect.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+
+# This script takes as input the shared folder. So for example you should called it like
+# ./pcluster_connect.sh "/shared"
+
+# Return the sessionid, the port and the tokenid
+
+create_new_session() {
+    dcv_path=$1
+    shared_folder_path=$2
+
+    dcv_file="${dcv_path}/dcv_session"
+    sessionid=$(shuf -zer -n20  {A..Z} {a..z} {0..9})
+    echo ${sessionid} > ${dcv_file}
+    dcv create-session --type virtual --storage-root ${shared_folder_path} ${sessionid}
+
+    echo ${sessionid}
+}
+
+main() {
+    shared_folder_path=$1
+    user=$(whoami)
+    os=$(cat /tmp/dna.json | jq -r .cfncluster.cfn_base_os)
+
+    if [[ ${os} != "centos7" ]]; then
+        >&2 echo "Non supported os"
+        exit 1
+    fi
+
+    if (( $(systemctl status dcvserver | grep inactive | wc -l) == 1 )); then
+        >&2 echo "DCV is not active on the given instance"
+        exit 1
+    fi
+
+    dcv_path="${HOME}/.parallelcluster_dcv"
+    mkdir -p ${dcv_path}
+
+    dcv_file="${dcv_path}/dcv_session"
+    if [[ ! -e ${dcv_file} ]]; then
+        sessionid=$(create_new_session ${dcv_path} ${shared_folder_path})
+    else
+        sessionid=$(cat ${dcv_file})
+
+        # number of session can either be 0 or 1
+        number_of_session=$(dcv list-sessions |& grep ${user} | grep ${sessionid} | wc -l)
+        if (( ${number_of_session} == 0 )); then
+            # The system has been rebooted so we need to create a new session cause of possible DOS
+            sessionid=$(create_new_session ${dcv_path} ${shared_folder_path})
+        fi
+    fi
+
+    tokenid=$(shuf -zer -n20  {A..Z} {a..z} {0..9})
+    echo ${tokenid} | dcvsimpleextauth add-user --user ${user} --session ${sessionid}
+
+    # xargs to remove eventual whitespaces
+    port=$(cat /etc/dcv/dcv.conf | grep web-port= | awk -F'=' '{ print $2 }' | xargs)
+    echo "${sessionid} ${port} ${tokenid}"
+}
+
+if [[ -z "$1" ]]; then
```

Comment:
we should create a _parse_options function, to be called as first thing in the main